hyperspy/io_plugins/emd.py

```diff
@@ -621,25 +668,37 @@ def _read_image(self, image_group, image_sub_group_key):
             self.detector_name = original_metadata['Detectors']['Detector-01']['DetectorName']
 
         read_stack = (self.load_SI_image_stack or self.im_type == 'Image')
-        if read_stack:
-            data = np.rollaxis(np.array(image_sub_group['Data']), axis=2)
-            # Get the scanning area shape of the SI from the images
-            self.SI_shape = data.shape[1:]
+        h5data = image_sub_group['Data']
+        # Get the scanning area shape of the SI from the images
+        self.spatial_shape = h5data.shape[:-1]
+        # Set the axes in frame, y, x order
+        if self.lazy:
+            if self.lazy:
+                data = da.transpose(
+                    da.from_array(
+                        h5data,
+                        chunks=h5data.chunks),
+                    axes=[2, 0, 1])
         else:
-            data = image_sub_group['Data'][:, :, 0]
-            # Get the scanning area shape of the SI from the images
-            self.SI_shape = data.shape
+            data = np.rollaxis(np.array(h5data), axis=2)
```

Comment:
Accessing the hdf5 dataset is slow, possibly because there are as many chunks as images in the stack.
For the case of not loading the SI image stack, I would read only the first chunk and change accordingly below, since it is much faster (more than 10x):
``` python
if not read_stack:
    data = h5data[:, :, 0]
else:
    data = np.rollaxis(np.array(h5data), axis=2)
```

Reply:
Actually, I wanted to discuss with you this feature. Wouldn't it be better to remove ``read_stack`` all together? If the user is not loading lazily and the SI fits in memory, it's likely that the image stack fits too. If loading lazily, it doesn't matter at all. What do you think?

Reply:
I think this is still useful. We don't want to always load the stack of images and there are cases, where users may want to load the stack of images but not the stack of SI: for example, to check if there were damage/contamination/drift visible on the image.
However, it would be good to tweak the default: when we load the stack of SI, the default of `load_SI_image_stack` should be `True` and otherwise it should be 'False'.

Reply:
What about not loading any image of the image stack (instead of just the first one) when ``load_SI_image_stack`` is ``False``? (Note that we can get the shape without actually loading the image. Also, we maybe should get the shape from elsewhere to consider the case of a SI acquired without recording the image.)

We could also rename the kwarg to ``load_SI_image``.

Reply:
If loading the whole stack was as fast as loading the first image, I could prefer your suggestion, also renaming to `load_SI_image` would be more clear with a default to `True`. But it can take a couple of seconds to load the whole stack so I find it a bit annoying.